### PR TITLE
EOS-27251: Support Bundle filter implementation

### PR DESCRIPTION
**Pull request by rohitkolapkar**
Signed-off-by: rohitkolapkar <rohit.j.kolapkar@seagate.com>

# Problem Statement
[EOS-27251:](https://jts.seagate.com/browse/EOS-27251) [CSM] Add filter support for support bundle

# Design
-  For Feature, Post the link for design
https://seagate-systems.atlassian.net/wiki/spaces/PUB/pages/749076684/Node+level+Support+Bundle+Collection#Future-implementations-(PI-5)
As per the document,
```
#NOTE: Below are the additional arguments, that all component specific Support Bundle script should accept.
    # --duration -> ISO 8061 format, "P" separates, the start datetime and duration to be considered after that. "T" separated date and time. Default is P5d.
    # --size_limit -> Limit set on per component for its support bundle size. eg. "1G" for 1GB or "100M" for 100 MB. Default -> 500MB per node.
```

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [x] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [x] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [x] Dependencies on other component(s)
Found two issues while consuming py-utils SB filter API:- [EOS-27275](https://jts.seagate.com/browse/EOS-27275)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

# Unit Testing Screenshots
1. CSM Support Bundle is now accepting all the filter args 
image

2. No regression, support bundle generation is working fine
image

image

image

image

### CODE REVIEW

File: csm/cli/support_bundle/csm_bundle_generate.py [hunk 1]
```diff
@@ -111,16 +113,58 @@ class GenerateCsmBundle:
     '''
     @staticmethod
     def generate_bundle(args):
+
         Conf.load(const.CONSUMER_INDEX, args['config_url'])
         log_path = Conf.get(const.CONSUMER_INDEX, const.CSM_LOG_PATH_KEY)
-        csm_log_path = os.path.join(log_path,'csm')
+        csm_log_path = os.path.join(log_path, const.CSM_COMPONENT_NAME)
         bundle_id = args['bundle_id']
         target_path = args ['target']
```
Comment (pawankumarsrivastava) [inline csm/cli/support_bundle/csm_bundle_generate.py:121]: Move static strings to const file

File: csm/cli/support_bundle/csm_bundle_generate.py [hunk 2]
```diff
@@ -111,16 +113,58 @@ class GenerateCsmBundle:
...
-        services = args['services'] #Not making any use of service option for now
-        target_path = os.path.join(target_path, bundle_id)
+        duration = args['duration']
+        size_limit = args['size_limit']
+        # services = args['services'] #Not making any use of service option for now
```
Comment (pawankumarsrivastava) [inline csm/cli/support_bundle/csm_bundle_generate.py:124]: Remove the comment

File: csm/cli/support_bundle/csm_bundle_generate.py [hunk 3]
```diff
@@ -111,16 +113,58 @@ class GenerateCsmBundle:
...
+
+        csm_size_filtered_logs_dir = os.path.join(const.CSM_SETUP_LOG_DIR,
+            f"{const.CSM_COMPONENT_NAME}_logs_size")
+        GenerateCsmBundle.__clear_tmp_files(csm_size_filtered_logs_dir)
+        os.makedirs(csm_size_filtered_logs_dir,exist_ok=True)
+        csm_time_filtered_logs_dir = os.path.join(const.CSM_SETUP_LOG_DIR,
+            f"{const.CSM_COMPONENT_NAME}_logs_size_time")
+        GenerateCsmBundle.__clear_tmp_files(csm_time_filtered_logs_dir)
+        os.makedirs(csm_time_filtered_logs_dir,exist_ok=True)
+        FilterLog.limit_size(csm_log_path, csm_size_filtered_logs_dir,
```
Comment (pawankumarsrivastava) [inline csm/cli/support_bundle/csm_bundle_generate.py:139]: Comment this and add TODO.

**Review by pawankumarsrivastava — CHANGES_REQUESTED** [review]

File: csm/cli/support_bundle/csm_bundle_generate.py (hunk 1 above)
Comment (rohitkolapkar) [reply csm/cli/support_bundle/csm_bundle_generate.py]: fixed

File: csm/cli/support_bundle/csm_bundle_generate.py (hunk 2 above)
Comment (rohitkolapkar) [reply csm/cli/support_bundle/csm_bundle_generate.py]: By removing the comments coday is failing for unused vars. So as u suggested initialized all the vars at class level.
Fixed.

File: csm/cli/support_bundle/csm_bundle_generate.py (hunk 3 above)
Comment (rohitkolapkar) [reply csm/cli/support_bundle/csm_bundle_generate.py:149]: Done.
Commented out duration filter as we have a open bug - 
[Support Bundle - Filter logs based on duration not working properly](https://jts.seagate.com/browse/EOS-27275)

**Review by shrimetta — APPROVED** [review]

**Review by soniyamoholkar — APPROVED** [review]

**Review by udayan-y21 — APPROVED** [review]

**Review by pawankumarsrivastava — APPROVED** [review]

**Pull request merged**